ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@JSStats, you are now talking about a language extension again, I wasn't. But even with more general subtyping like you propose, you wouldn't need or want to adjust tuples in the operational semantics. In a tuple-based implementation, subtyping just works, without coercing, because there is no operation that can ever observe excess values. They just get ignored when decomposing a tuple. A real-world implementation might of course want to implement that differently, e.g. a stack-based one has to.